botocore/handlers.py

```diff
@@ -110,10 +110,34 @@ def json_decode_template_body(parsed, **kwargs):
 def calculate_md5(params, **kwargs):
     request_dict = params
     if request_dict['body'] and 'Content-MD5' not in params['headers']:
-        md5 = hashlib.md5()
-        md5.update(six.b(params['body']))
-        value = base64.b64encode(md5.digest()).decode('utf-8')
-        params['headers']['Content-MD5'] = value
+        body = request_dict['body']
+        if isinstance(body, bytes):
+            binary_md5 = _calculate_md5_from_bytes(body)
+        else:
+            binary_md5 = _calculate_md5_from_file(body)
+        base64_md5 = base64.b64encode(binary_md5).decode('ascii')
```

Comment:
why ascii now? It looks like it used to be decoded with utf-8. I do not think it will make a difference though since it is just an md5.